packages/cli/src/commands/init/utils/repo.ts

```diff
@@ -41,23 +42,44 @@ export const cloneRepo = async (designSystemOptions: any): Promise<boolean> => {
             return false;
         }
 
-        const spinner = ora(chalk.hex('#d44527').bold('Initializing design system repository and cloning from remote...')).start();
-
         const dir = path.join(process.cwd(), designSystemOptions['ds-name']);
-        await git.clone({
-            fs,
-            dir,
-            http: gitHttp,
-            url: `https://github.com/${ repoRef.repo }`,
-            depth: 1,
-        });
-        await git.checkout({
-            fs,
-            dir,
-            ref: repoRef.branch,
-        });
 
-        spinner.stop();
+        const gitSetup = [
+            {
+                title: 'Download',
+                task: (_ctx:any,task:any) => git.clone({
+                    fs,
+                    dir,
+                    http: gitHttp,
+                    url: `https://github.com/${ repoRef?.repo }`,
```

Comment:
What's the reason for optional chaining `?.`? `repoRef` should already be guaranteed not-null by the `if` above:
https://github.com/rangle/radius-workspace/blob/55ae2fdbb5870f4ab1fab306a7eeddc6f535d295/packages/cli/src/commands/init/utils/repo.ts#L40-L43

Same for line 68 below.

Reply:
Typescript threw errors saying it could be null, adding the ?. removed the error.
Would setting the initial value of repoRef to branchRef('', '') be more appropriate, and change the condition to check repo string?

Reply:
Interesting. Apparently, `repoRef` type gets re-evaluated within the function. And surprisingly, assigning the variable achieves the desired type:
```
const definitiveRef = repoRef;
const gitSetup = [
    ...
    url: `https://github.com/${ definitiveRef.repo }`,
    ...
];
```

Reply:
I've made this change 👍